rasa/cli/data.py

```diff
@@ -109,4 +149,51 @@ def validate_files(args) -> NoReturn:
         sys.exit(1)
 
     everything_is_alright = validator.verify_all(not args.fail_on_warnings)
-    sys.exit(0) if everything_is_alright else sys.exit(1)
+    if not args.max_history:
+        logger.info(
+            "Will not test for inconsistencies in stories since "
+            "you did not provide a value for `--max-history`."
+        )
+    else:
+        # Only run story structure validation if everything else is fine
+        # since this might take a while
+        everything_is_alright = validator.verify_story_structure(
+            not args.fail_on_warnings, max_history=args.max_history
+        )
+
+    if not everything_is_alright:
```

Comment:
also this would be good to be covered in tests (same for the `validate_stories`)